### PR TITLE
Feature: support ellipse commands

### PR DESCRIPTION
**Pull request by kyle1320**
## Feature

Currently, ellipse commands aren't supported due to lack of support for the `ellipse()` canvas method in IE. However, ellipses can also be drawn by transforming the canvas and using the `arc()` command, which is widely supported.

This patch adds support for `Path2D.ellipse()`, as well as full support for SVG arc commands (`a` and `A`) including those with different x and y radii and rotation.

The code to convert from endpoint to center parameterization was based off of the W3 SVG implementation notes which can be found at https://www.w3.org/TR/SVG/implnote.html#ArcImplementationNotes .

Also includes automated tests and an additional visual example.

### CODE REVIEW

**Comment by skreborn** [comment]
@nilzona Sorry to bother, but is there any chance you could take a look at this PR? This polyfill would be a tremendous help for us with an upcoming feature, but this change is absolutely required to make it work properly. I can make a fork and pull it in myself in the meantime as a temporary solution, but I'd love for it to be officially part of the project.

File: test/path2d-polyfill.spec.js [hunk 1]
```diff
@@ -195,6 +200,15 @@ describe('Canvas path', () => {
         ctx.stroke(p);
         cMock.verify();
       });
+      it('ellipse', () => {
+        cMock.expects('translate').once().withArgs(100, 150);
+        cMock.expects('scale').once().withArgs(20, 40);
+        cMock.expects('arc').once().withArgs(0, 0, 1, 0, Math.PI, true);
+        const p = new window.Path2D();
+        p.ellipse(100, 150, 20, 40, 0, Math.PI, true);
```
Comment (skreborn) [inline test/path2d-polyfill.spec.js]:
```suggestion
        p.ellipse(100, 150, 20, 40, 0, 0, Math.PI, true);
```

File: example/test3.js [hunk 2]
```diff
@@ -50,3 +50,11 @@ path5.closePath();
 ctx.fillStyle = 'rgba(255, 255, 0, 0.8)';
 ctx.stroke(path5);
 ctx.fill(path5)
+
+ctx.translate(0, 50);
+var path5 = new Path2D();
```
Comment (skreborn) [inline example/test3.js]: Use the name `path6` for consistency, and include the missing `angle` parameter (more on that later).

```javascript
var path6 = new Path2D();
path6.ellipse(150, 120, 60, 40, 0, 0, Math.PI, true);
path6.closePath();
ctx.fillStyle = 'rgba(255, 0, 0, 0.8)';
ctx.stroke(path6);
ctx.fill(path6)
```

File: test/path2d-polyfill.spec.js [hunk 3]
```diff
@@ -265,18 +279,30 @@ describe('Canvas path', () => {
...
+        cMock.expects('scale').once().withArgs(45, 45);
+        cMock.expects('arc').once().withArgs(0, 0, 1, Math.PI, Math.PI / 2, false);
         ctx.stroke(new window.Path2D('M230 230a 45 45 0 1 1 45 45L 275 230 Z'));
         cMock.verify();
       });
+
+      it('a - elliptical', () => {
+        cMock.expects('translate').once().withArgs(250, 230);
+        cMock.expects('scale').once().withArgs(20, 40);
+        cMock.expects('arc').once().withArgs(0, 0, 1, Math.PI, 0, false);
```
Comment (skreborn) [inline test/path2d-polyfill.spec.js]: The tests are complaining that this should be negative zero for some reason beyond my comprehension.

```suggestion
        cMock.expects('arc').once().withArgs(0, 0, 1, Math.PI, -0, false);
```

**Comment by nilzona** [comment]
Hey @kyle1320  and @skreborn. Thank you for submitting this PR and I apologize for taking so much time to answer. I don't know why I missed this PR ... must have been because I get ~50+ notification e-mails from github every day and I didn't expect this PR to appear :)  I will take a look at it and try to take it in.

File: example/test3.js [hunk 4]
```diff
@@ -50,3 +50,11 @@ path5.closePath();
...
+path5.ellipse(150, 120, 60, 40, 0, Math.PI, true);
```
Comment (kyle1320) [inline example/test3.js]:
```suggestion
path5.ellipse(150, 120, 60, 40, 0, 0, Math.PI, true);
```

File: test/path2d-polyfill.spec.js (hunk 3 above)
Comment (kyle1320) [reply test/path2d-polyfill.spec.js]: I'm not seeing this behavior, so I'm hesitant to commit it. Both `0` and `-0` pass on my machine.

File: example/test3.js (hunk 2 above)
Comment (kyle1320) [reply example/test3.js]: Done. Thanks for the review!

File: test/path2d-polyfill.spec.js (hunk 3 above)
Comment (kyle1320) [reply test/path2d-polyfill.spec.js]: Ah never mind, my dependencies were out of date. An update to sinon (via after-work.js) caused this to fail.

**Pull request merged**